[Chen Xiangrui] iP

chen-xiangrui opened the pull request:
# RuiBot
> “Drive thy business or it will drive thee.” – Benjamin Franklin ([source](https://dansilvestre.com/productivity-quotes/#5_%E2%80%9CDrive_thy_business_or_it_will_drive_thee%E2%80%9D_%E2%80%93_Benjamin_Franklin))

RuiBot helps you remember so that you don't have to. It is:
- Simple to use
- Easy to navigate
- Efficient way of recording your events

To get started:
1. Download the .jar v1.0 file.
2. Double-click and open the file.
3. Add tasks and start using.

And... _let's go!_ :+1:

> [!IMPORTANT]
> Make sure to stick to the correct format for input of tasks!

> [!TIP]
> If you are using gradle to run, use `./gradlew run`.

Current features:
- [x] Manage different type of tasks
- [x] Simple CLI integration for ease of use
- [ ] GUI feature

To show more insights on how the bot works, this is how the bot is linked with a storage textfile which stores all the tasks:
```java
    public ArrayList<String> load() {
        ArrayList<String> lines = new ArrayList<>();
        while (this.scanner.hasNext()) {
            String line = this.scanner.nextLine();
            lines.add(line);
        }
        return lines;
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

chen-xiangrui committed:
Rename, Greet, Exit

chen-xiangrui committed:
Revert "Rename, Greet, Exit"

This reverts commit 4feca888b42ec48c63ad7b41e7aeeb806f6c9830.

chen-xiangrui committed:
Echo

chen-xiangrui committed:
Merge branch 'master' of https://github.com/chen-xiangrui/ip

# Conflicts:
#	src/main/java/Duke.java

chen-xiangrui committed:
Add, List

chen-xiangrui committed:
Mark as Done

chen-xiangrui committed:
ToDos, Events, Deadlines

chen-xiangrui committed:
Automated Text UI Testing

chen-xiangrui committed:
Handle Errors

chen-xiangrui committed:
Delete

chen-xiangrui committed:
Update RuiBot.java

Let's,
* Add new line space to facilitate testing of pull requests

chen-xiangrui committed:
Updated RuiBot.java, Task.java, ToDo.java, Deadline.java, Event.java

Let's,
* Add in logging functions to store tasks in ruibot.txt
* Add fields isCompleted to Task constructor
* Add parameter isCompleted, isStart to addItems in RuiBot

chen-xiangrui committed:
Updated RuiBot.java

Let's,
* Change startDate and endDate format to accept "yyyy-MM-dd HHmm" format, and input "MMM dd yyyy, h:mm a" format

chen-xiangrui committed:
Added Ui.java, Storage.java, Parser.java, TaskList.java, updated RuiBot.java

Let's:
* Handle all interactions with user using Ui class
* Use Storage class to handle loading and storing of tasks in ruibot.txt
* Use Parser class to categorise the inputs into their respective command type
* Use TaskList class to encapsulate all tasks and perform task methods
* Based on the changes made above, remove the relevant blocks of code and methods in RuiBot.java

chen-xiangrui committed:
Added packages: ruibot, ruibot.tasks

Let's:
* Package all classes under ruibot package
* Under ruibot package, create tasks package to add all Task classes: Task, ToDo, Deadline, Event

chen-xiangrui committed:
Merge branch 'add-gradle-support'

chen-xiangrui committed:
Updated build.gradle

Let's:
* Change the entry point of Gradle to ruibot.RuiBot, such that Gradle can run RuiBot

chen-xiangrui committed:
Added ParserTest.java, TaskListTest.java

Let's:
* Add in JUnit tests to test functionality of methods to reduce bugs by implementing ParserTest to test Parser class's methods and TaskListTest to test TaskList class's methods, and run the test cases on gradle.

chen-xiangrui committed:
Updated build.gradle

Let's:
* Change shadowJar.archiveBaseName to ruibot to edit file name of .jar file

chen-xiangrui committed:
Updated RuiBot.java, Storage.java, Parser.java, TaskList.java

Let's:
* Add JavaDocs statements to the classes and methods of RuiBot, Storage, Parser, TaskList.

chen-xiangrui committed:
Updated Storage.java, TaskList.java, ToDo.java, ParserTest.java, TaskListTest.java

Let's:
* Comply with Java coding standard and update the files accordingly.
* Ensure consistent ordering of imported packages in each file.

chen-xiangrui committed:
Updated RuiBot.java, TaskList.java, Task.java

Let's:
* Add find() method in TaskList, to print the tasks containing a keyword provided.
* Update mark() and unmark() methods in TaskList, to handle exceptions where user wishes to mark a completed task and unmark an uncompleted task.
* Add contains() method in Task, to return a boolean of whether the Task contains the provided keyword.
* Add isTaskCompleted() method in Task, to return a boolean on whether the Task is completed.
* Update main() method in RuiBot, to handle new command type "find".

chen-xiangrui committed:
Merge branch 'branch-A-CodingStandard'

chen-xiangrui committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/ruibot/TaskList.java

botosek commented on src/main/java/ruibot/Storage.java:
```diff
@@ -0,0 +1,66 @@
+package ruibot;
+
+import java.io.IOException;
+
+import ruibot.tasks.Task;
+
+import java.io.File;
+import java.io.FileWriter;
+
+import java.util.ArrayList;
+
+import java.util.Scanner;
```
I like that some of the import statements are grouped together which helps make it more organized! Perhaps the import java.* statements could be grouped together to make it more organized and standardized?

botosek commented on src/main/java/ruibot/RuiBot.java:
```diff
@@ -0,0 +1,72 @@
+                    if (command.equals("bye")) {
+                        ruibot.ui.goodbyeMessage();
+                        break;
+                    } else if (command.equals("list")) {
+                        ruibot.tasks.printList();
+                    } else if (command.equals("mark")) {
+                        int index = Integer.parseInt(input.substring(5));
+                        ruibot.tasks.markItem(index);
+                        ruibot.storage.save(ruibot.tasks.getTasks());
+                    } else if (command.equals("unmark")) {
+                        int index = Integer.parseInt(input.substring(7));
+                        ruibot.tasks.unmarkItem(index);
+                        ruibot.storage.save(ruibot.tasks.getTasks());
+                    } else if (command.equals("todo") || input.startsWith("deadline") || input.startsWith("event")) {
+                        ruibot.tasks.addItem(input, false, false);
+                        ruibot.storage.save(ruibot.tasks.getTasks());
+                    } else if (command.equals("delete")) {
+                        int index = Integer.parseInt(input.substring(7));
+                        ruibot.tasks.deleteItem(index);
+                        ruibot.storage.save(ruibot.tasks.getTasks());
+                    } else if (command.equals("find")) {
+                        ruibot.tasks.find(input.substring(5));
```
I like how all the commands except the individual add task commands are parsed out together!

botosek commented on src/main/java/ruibot/tasks/ToDo.java:
```diff
@@ -0,0 +1,14 @@
+        super(name, isCompleted);
+    }
+
+    public String taskString(){
```
Perhaps there should be a white space between the closing bracket and the opening egyptian brace?

botosek reviewed (commented):
LGTM! Just a few nits.

Partillay commented on src/main/java/ruibot/tasks/Event.java:
```diff
@@ -0,0 +1,18 @@
+        this.endDate = endDate;
+    }
+
+    public String taskString(){
```
Perhaps there should be a space between ) and {?

Partillay reviewed (commented):
LGTM. The code looks neat :). Just very little formatting flaws to fix!

johngao122 commented on src/main/java/Deadline.java:
```diff
@@ -1,8 +1,8 @@
-    public Deadline(String name, String endDate) {
-        super(name);
+    public Deadline(String name, boolean isCompleted, String endDate) {
+        super(name, isCompleted);
```
Nice that you kept the state in the task itself, good job!

johngao122 commented on src/main/java/RuiBot.java:
```diff
@@ -37,16 +41,20 @@ public void addItem(String input) throws EmptyTaskException {
+                    + task.taskString() + "\n"
+                    + "Now you have " + this.itemsList.size() + " tasks in the list.\n"
+                    + "____________________________________________________________\n");
+        }
```
Have you considered wrapping this in a UI class

johngao122 commented on src/main/java/RuiBot.java:
```diff
@@ -96,11 +122,53 @@ public static void main(String[] args) {
+            for (String line : lines) {
+                String item;
+                boolean isCompleted = (line.charAt(5) == 'X');
+                if (line.charAt(1) == 'T') {
```
Could fail with malformed input. Consider more robust parsing.

johngao122 reviewed (commented):
Overall clear variable naming and organization for your logic! Consider refactoring some parts to make the code more robust and maintainable.

chen-xiangrui committed:
Add ruibot.gui package, DialogBox.java, Launcher.java, Main.java, MainWindow.java
Add resources/images directory, DaDuke.png, DaUser.png
Add resources/view directory, MainWindow.fxml, DialogBox.fxml
Update RuiBot.java, TaskList.java, Ui.java

Let's:
* Implement GUI in ruibot.gui package with Main, MainWindow, Launcher, DialogBox
* Contain ImageView elements with images in resources/images directory
* Do styling via FXML in resources/view directory for MainWindow and DialogBox
* Create getResponse() method in RuiBot.java to create response for RuiBot to return to user
* Update methods in TaskList to return String instead of void methods

chen-xiangrui committed:
Update RuiBot.java

Let's:
* Remove commented out main() method as it is no longer in use

chen-xiangrui committed:
Update build.gradle

Let's,
* edit the version of the .jar file for the creating of the new version of RuiBot

chen-xiangrui committed:
Update TaskList.java and MainWindow.java

Let's,
* implement assertions on addItem(), markItem(), unmarkItem()
in TaskList, and handleUserInput() in MainWindow.

chen-xiangrui committed:
Merge pull request #2 from chen-xiangrui/branch-A-Assertions

Update TaskList.java and MainWindow.java

chen-xiangrui committed:
Update TaskList.java

Let's,
* make modifications to addItem() method in TaskList to improve code quality,
integrating SLAP principles, reducing method lengths, remove complicated expressions.

chen-xiangrui committed:
Merge pull request #3 from chen-xiangrui/branch-A-CodeQuality

Update TaskList.java

chen-xiangrui committed:
Update Ruibot.java, TaskList.java, Deadline.java, Event.java,
ToDo.java, Task.java

Let's,
* implement new feature: ViewSchedules.
* create new abstract method containsDate() in Task.
* implement respective containsDate() for Task's subclasses.
* implement schedule() method to handle schedule output for that day.

chen-xiangrui committed:
Update RuiBot.java

Let's,
* change some of the output messages.

chen-xiangrui committed:
Update Parser.java, RuiBot.java, Storage.java, TaskList.java,
Ui.java, EmptyTaskException.java, WrongInputException.java,
ToDo.java, Deadline.java, Event.java, Task.java, Main.java,
DialogBox.java

Let's,
* implement Javadocs for the classes and its methods

chen-xiangrui committed:
Add RuiBotTest.java

Let's,
* add a JUnit testing method for RuiBot.java

chen-xiangrui committed:
Update docs/README.md

Let's,
* update the README with important details for users to know

chen-xiangrui committed:
Update Ui.png

Let's:
* change file type from jpg to png

chen-xiangrui committed:
Update build.gradle

Let's:
* update the version of new jar file being generated

chen-xiangrui committed:
Update Storage.java

Let's,
* fix the issue of not being able to locate filepath

chen-xiangrui committed:
Update RuiBot.java, Storage.java, Main.java

Let's,
* pass in directory and file name as arguments separately instead

chen-xiangrui committed:
Update MainWindow.java

Let's,
* add a welcome message at the start
